terms.lua

```diff
@@ -1368,82 +1370,167 @@ local function replace_flex_values(tag, v)
+		if arg_value_t == strict_value then
+			---@cast arg ArrayValue<strict_value>
+			return U.notail(arg:map(array(flex_value), flex_value.strict))
+		elseif arg_value_t == stuck_value then
+			---@cast arg ArrayValue<stuck_value>
+			return U.notail(arg:map(array(flex_value), flex_value.stuck))
+		end
+	elseif getmetatable(getmetatable(arg)) == gen.map_type_mt then
+		---@cast arg MapValue
+		local arg_mt = getmetatable(arg) --[[@as MapType]]
+		local arg_key_t, arg_value_t = arg_mt.key_type, arg_mt.value_type
+		if
+			arg_key_t == strict_value
+			or arg_key_t == stuck_value
+			or arg_value_t == strict_value
+			or arg_value_t == stuck_value
+		then
+			local arg_flex_key_t, transform_arg_key = arg_key_t
+			if arg_key_t == strict_value then
+				arg_flex_key_t, transform_arg_key = flex_value, flex_value.strict
+			elseif arg_key_t == stuck_value then
+				arg_flex_key_t, transform_arg_key = flex_value, flex_value.stuck
+			else
+				function transform_arg_key(key)
+					return key
+				end
+			end
+			local arg_flex_value_t, transform_arg_value = arg_value_t
+			if arg_value_t == strict_value then
+				arg_flex_value_t, transform_arg_value = flex_value, flex_value.strict
+			elseif arg_value_t == stuck_value then
+				arg_flex_value_t, transform_arg_value = flex_value, flex_value.stuck
+			else
+				function transform_arg_value(value)
+					return value
+				end
 			end
-			table.insert(flex_args, flex_array)
-		elseif array(stuck_value).value_check(v) then
-			local flex_array = array(flex_value)()
-			for _, v in ipairs(v) do
-				flex_array:append(flex_value.stuck(v))
+			local arg_flex_map = {}
+			for arg_key, arg_value in pairs(arg._map) do
+				arg_flex_map[transform_arg_key(arg_key)] = transform_arg_value(arg_value)
 			end
-			table.insert(flex_args, flex_array)
-		elseif strict_continuation.value_check(v) then
-			table.insert(flex_args, flex_continuation.strict(v))
-		elseif stuck_continuation.value_check(v) then
-			table.insert(flex_args, flex_continuation.stuck(v))
-		elseif strict_runtime_context_type.value_check(v) then
-			table.insert(flex_args, v:as_flex())
-		else
-			table.insert(flex_args, v)
+			return U.notail(map(arg_flex_key_t, arg_flex_value_t):unchecked_new(arg_flex_map))
 		end
+	elseif strict_continuation.value_check(arg) then
+		---@cast arg strict_continuation
+		return U.notail(flex_continuation.strict(arg))
+	elseif stuck_continuation.value_check(arg) then
+		---@cast arg stuck_continuation
+		return U.notail(flex_continuation.stuck(arg))
+	elseif strict_runtime_context_type.value_check(arg) then
+		---@cast arg StrictRuntimeContext
+		return U.notail(arg:as_flex())
 	end
+	return arg
+end
 
+---@param args (Value | StrictRuntimeContext | FlexRuntimeContext)[]
+---@return (Value | FlexRuntimeContext)[] arg
+local function unify_flex_values(args)
+	---@type (Value | FlexRuntimeContext)[]
+	local flex_args = {}
+	for i, v in ipairs(args) do
+		flex_args[i] = unify_flex_value(v)
```

Comment:
There doesn't seem to be any actual reason to create another unify_flex_value function just to iterate over it

Reply:
Partially to align with `specify_flex_value()`/`specify_flex_values()`, and partially to benefit from early returns. Though, `unify_flex_type()` is taking all those early returns now instead of `unify_flex_value()`.